File: go-solutions/735_asteroid_collision_test.go
package leetcode

import (
	"go-solutions/utils"
	"testing"
)

type asteroidCollisionTest struct {
	asteroids []int
	want      []int
}

var asteroidCollisionTests = []asteroidCollisionTest{
	{[]int{5, 10, -5}, []int{5, 10}},
	{[]int{8, -8}, []int{}},
	{[]int{10, 2, -5}, []int{10}},
}

func TestAsteroidCollision(t *testing.T) {
	for _, test := range asteroidCollisionTests {
		got := AsteroidCollision(test.asteroids)
		if utils.SliceEqual(got, test.want) == false {
			t.Errorf("passed = %v, got = %v, want = %v", test.asteroids, got, test.want)
		}
	}
}

File: ts-solutions/tests/735-asteroid-collision.test.ts
import { asteroidCollision } from '../src/735-asteroid-collision'

type AsteroidCollisionTest = {
	asteroids: number[]
	want: number[]
}

const asteroidCollisionTests: AsteroidCollisionTest[] = [
	{ asteroids: [5, 10, -5], want: [5, 10] },
	{ asteroids: [8, -8], want: [] },
	{ asteroids: [10, 2, -5], want: [10] },
	{ asteroids: [-2, -2, 1, -2], want: [-2, -2, -2] },
	{ asteroids: [-2, 1, 1, -2], want: [-2, -2] },
]

describe(`asteroidCollision`, () => {
	it.each(asteroidCollisionTests)(
		`asteroidCollision($asteroids) => $want`,
		({ asteroids, want }) => {
			const got = asteroidCollision(asteroids)
			expect(got).toEqual<number[]>(want)
		}
	)
})
